### PR TITLE
Ease of use improvements for Chart RPC WIP #2418

### PR DESCRIPTION
**Pull request by gsteinLTU**
#2418 and sort of a prereq for #2417.

The big change is allowing lists of numbers to be plotted as a line chart, filling in the x coordinates sequentially. Need to figure out if this creates any new error cases.

### CODE REVIEW

**Comment by hamidzr** [comment]
note for later: remember to run at least the following to see if we are still backward compatible:
1. `src/browser/test/rpc/chart-tests.xml`
2. eclipse visualization project

**Comment by gsteinLTU** [comment]
`src/browser/test/rpc/chart-tests.xml` seems to be working about the same as on editor (so it's not any more broken than it already is). The part getting the data for the stations doesn't appear to work on either.

I can't get the eclipse service to work at all on my machine, so that may require you to test. This may be a bigger issue: https://apicommunity.wunderground.com/weatherapi/topics/end-of-service-for-the-weather-underground-api.

File: src/server/rpc/procedures/chart/chart.js [hunk 1]
```diff
@@ -1,15 +1,15 @@
 /**
- * Charting service powered by Gnuplot
+ * Charting service powered by gnuplot
  *
  * @service
  */
 
-const ApiConsumer = require('../utils/api-consumer'),
+const NBService = require('../utils/service'),
```
Comment (hamidzr) [inline src/server/rpc/procedures/chart/chart.js]: This is a better fit :+1:

File: src/server/rpc/procedures/chart/chart.js [hunk 2]
```diff
@@ -66,23 +66,33 @@ function calcRanges(lines, isCategorical){
 }
 
 function prepareData(input, options) {
-    // if the input is one line convert it to appropriate format
     const xShouldBeNumeric = !options.isCategorical && !options.isTimeSeries;
-
-    if (! Array.isArray(input[0][0])){
+    
+    // if the input is one line convert it to appropriate format
+    if (Array.isArray(input) && !Array.isArray(input[0]) || !Array.isArray(input[0][0]) && input[0].length === 2){
```
Comment (hamidzr) [inline src/server/rpc/procedures/chart/chart.js]: the main changes for this PR happen here and a few lines down where you check for 
>> // If only one dimension is given
and the rest are formatting updates correct?

**Comment by hamidzr** [comment]
the eclipse service is using data that is stored in the database which we don't have locally. You could potentially set dev to run off of this branch to test it.

File: src/server/rpc/procedures/chart/chart.js (hunk 2 above)
Comment (gsteinLTU) [reply src/server/rpc/procedures/chart/chart.js]: Yeah, that's the only major change.

**Pull request merged**